service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -47,16 +48,31 @@ public SmartRunsPoller(
     this.objectMapper = objectMapper;
   }
 
-  public RecordResponse updateOutputAttributes(
-      Run run, List<WorkflowOutputDefinition> outputDefinitionList) throws Exception {
+  public boolean hasOutputDefinition(Run run) throws JsonProcessingException {
+    List<WorkflowOutputDefinition> outputDefinitionList =
+        objectMapper.readValue(run.runSet().method().outputDefinition(), new TypeReference<>() {});
+
+    return !outputDefinitionList.isEmpty();
+  }
+
+  public void updateOutputAttributes(Run run)
+      throws WorkflowOutputNotFoundException, ApiException, JsonProcessingException,
+          org.databiosphere.workspacedata.client.ApiException {
+    List<WorkflowOutputDefinition> outputDefinitionList =
+        objectMapper.readValue(run.runSet().method().outputDefinition(), new TypeReference<>() {});
+
     Object outputs = cromwellService.getOutputs(run.engineId());
     RecordAttributes outputParamDef = OutputGenerator.buildOutputs(outputDefinitionList, outputs);
     RecordRequest request = new RecordRequest().attributes(outputParamDef);
 
-    logger.info(
-        "Updating output attributes for Record ID {} from Run {}", run.recordId(), run.engineId());
+    if (logger.isDebugEnabled()) {
+      logger.debug(
+          "Updating output attributes for Record ID {} from Run {}.",
+          run.recordId(),
+          run.engineId());
+    }
```

Comment:
Sorry for flip flopping! Since these are just IDs, I think it's fine at `info` level. Also, the main reason for the `isDebugEnabled` is to avoid formatting strings that never get printed... but this debug method probably doesn't format the string if that isn't needed anyway, so it might be less necessary here...

Reply:
I will change it to info level. The reason I added `logger.isDebugEnabled()` was because it was giving me a code smell saying it will affect performance. Maybe there is another way to avoid code smell 🤔